security/Security.php

```diff
@@ -376,7 +376,8 @@ public function Link($action = null) {
 	 * sessions don't timeout. A common use is in the admin.
 	 */
 	public function ping() {
-		return 1;
+		if (Member::currentUserID()) return 1;
+		return Security::permissionFailure($this);
```

Comment:
Ok, there's a small problem with this... in that ping COULD be used outside of the CMS, such as in frontend forms, and this method throwing errors could cause breakages.

I suggest this change; You can still check if the user isn't logged in, but don't make it an error condition.

```php
public function ping() {
	$response = new SS_HTTPResponse("1");
		if(!Member::currentUserID() && Director::is_ajax() && CMSSecurity::enabled()) {
			$this->getResponse()->addHeader('X-Reauthenticate', '1');
		}
		return $response;
	}
```